### PR TITLE
Run yum clean all after package installation

### PR DESCRIPTION
**Pull request by aminvakil**
There are bunch of other best practices which can help reducing the final image size or setting pipefail and so.

But as this Dockerfile is not getting used in final product, I'm not sure if it's worth putting time into it.

### CODE REVIEW

**Comment by ilya-fedin** [comment]
From what I've seen, there's not yum cache in the final image, I assume base image has some hooks

**Comment by aminvakil** [comment]
I'm not sure, debian has this mechanism in its official docker images, but it has been advised to clean cache in centos official docker image: https://hub.docker.com/_/centos/

Also maybe because these cleaning happens in builder images.

**Comment by ilya-fedin** [comment]
So I rechecked
```
[root@4b5821fae407 tdesktop]# du -sh /var/cache              
178M    /var/cache
```

File: Telegram/build/docker/centos_env/Dockerfile [hunk 1]
```diff
@@ -22,15 +22,17 @@ ENV PATH ${PATH}:${QT5_PREFIX}/bin:${QT6_PREFIX}/bin
...
 
 RUN yum -y install git meson ninja-build autoconf automake libtool \
 	fontconfig-devel freetype-devel libX11-devel at-spi2-core-devel alsa-lib-devel \
 	pulseaudio-libs-devel mesa-libGL-devel mesa-libEGL-devel libudev-devel \
 	webkitgtk4-devel pkgconfig bison yasm file which xorg-x11-util-macros \
 	devtoolset-9-make devtoolset-9-gcc devtoolset-9-gcc-c++ \
 	devtoolset-9-binutils llvm-toolset-7.0 llvm-toolset-7.0-clang-devel \
-	llvm-toolset-7.0-llvm-devel
+	llvm-toolset-7.0-llvm-devel \
+ && yum clean all
```
Comment (ilya-fedin) [inline Telegram/build/docker/centos_env/Dockerfile:35]:
```suggestion
	llvm-toolset-7.0-llvm-devel perl-XML-Parser && yum clean all
```

File: Telegram/build/docker/centos_env/Dockerfile [hunk 2]
```diff
@@ -652,7 +654,8 @@ RUN rm -rf libsigcplusplus
 FROM patches AS glibmm
 COPY --from=mm-common ${LibrariesPath}/mm-common-cache /
 COPY --from=libsigcplusplus ${LibrariesPath}/libsigcplusplus-cache /
-RUN yum -y install perl-XML-Parser
+RUN yum -y install perl-XML-Parser \
+ && yum clean all
```
Comment (ilya-fedin) [inline Telegram/build/docker/centos_env/Dockerfile:658]:
```suggestion
```

File: Telegram/build/docker/centos_env/Dockerfile [hunk 3]
```diff
@@ -22,15 +22,17 @@ ENV PATH ${PATH}:${QT5_PREFIX}/bin:${QT6_PREFIX}/bin
 
 RUN yum -y install https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm
 RUN yum -y install https://packages.endpoint.com/rhel/7/os/x86_64/endpoint-repo-1.7-1.x86_64.rpm
-RUN yum -y install centos-release-scl
+RUN yum -y install centos-release-scl \
+ && yum clean all
```
Comment (ilya-fedin) [inline Telegram/build/docker/centos_env/Dockerfile:26]:
```suggestion
RUN yum -y install centos-release-scl  && yum clean all
```

File: Telegram/build/docker/centos_env/Dockerfile (hunk 2 above)
Comment (aminvakil) [reply Telegram/build/docker/centos_env/Dockerfile:656]: Remove `perl-XML-Parser` installation part completely?

Comment (ilya-fedin) [reply Telegram/build/docker/centos_env/Dockerfile:656]: It's moved to https://github.com/telegramdesktop/tdesktop/pull/17251/files#diff-6459e7429c8f96e46a7b23b51bf21be6b35242b1cfbe3da4d1c1dc1a898d08b9R33

**Comment by aminvakil** [comment]
Please tell me if I should squash my commits into one.

File: Telegram/build/docker/centos_env/Dockerfile [hunk 4]
```diff
@@ -22,15 +22,15 @@ ENV PATH ${PATH}:${QT5_PREFIX}/bin:${QT6_PREFIX}/bin
 
 RUN yum -y install https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm
 RUN yum -y install https://packages.endpoint.com/rhel/7/os/x86_64/endpoint-repo-1.7-1.x86_64.rpm
-RUN yum -y install centos-release-scl
+RUN yum -y install centos-release-scl  && yum clean all
```
Comment (ilya-fedin) [inline Telegram/build/docker/centos_env/Dockerfile:25]: Oops...
```suggestion
RUN yum -y install centos-release-scl && yum clean all
```

**Comment by ilya-fedin** [comment]
> Please tell me if I should squash my commits into one.

Yeah, please

**Comment by ilya-fedin** [comment]
> I'm not sure if it's worth putting time into it.

It would be nice to reduce it since GH action that tries to build the image fails due to free space

File: Telegram/build/docker/centos_env/Dockerfile [hunk 5]
```diff
@@ -20,17 +20,17 @@ ENV CMAKE_VER 3.21.3
 ENV CMAKE_FILE cmake-$CMAKE_VER-Linux-x86_64.sh
 ENV PATH ${PATH}:${QT5_PREFIX}/bin:${QT6_PREFIX}/bin
 
-RUN yum -y install https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm
-RUN yum -y install https://packages.endpoint.com/rhel/7/os/x86_64/endpoint-repo-1.7-1.x86_64.rpm
-RUN yum -y install centos-release-scl
+RUN yum -y install https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm && yum clean all
```
Comment (aminvakil) [inline Telegram/build/docker/centos_env/Dockerfile:23]: I'm not really sure about running `yum clean all` after these remote installations, but as I've stated in my commit message they may install some dependencies from official configured CentOS mirrors.

**Comment by aminvakil** [comment]
> > I'm not sure if it's worth putting time into it.
> 
> It would be nice to reduce the size since GH action that tries to build the image fails due to free space

Nice, I need to investigate this further then, do you want me to open a new PR later or wait until I come up with some other tweaks?

**Comment by ilya-fedin** [comment]
It's a question for @john-preston, he is the only who have merge rights

**Comment by john-preston** [comment]
Thanks! I think additional improvements can be made in separate PRs.

**Comment by aminvakil** [comment]
`hadolint` (https://github.com/hadolint/hadolint) shows lots of other warnings and errors, but I can think all of them can be skipped.

**Pull request merged**